libmamba/include/mamba/util/flat_bool_expr_tree.hpp

```diff
@@ -0,0 +1,738 @@
+// Copyright (c) 2023, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#ifndef MAMBA_UTIL_FLAT_EXPR_TREE_HPP
+#define MAMBA_UTIL_FLAT_EXPR_TREE_HPP
+
+#include <functional>
+#include <utility>
+#include <variant>
+#include <vector>
+
```

Comment:
Consider making the types constexpr as much as possibly allowed by the C++ version we are using (in C++23 they look like they could be completely usable at compile-time)

Reply:
The type are based on `std::vector`, which is not `constexpr` constructible until C++20.

Is there a point to make `constexpr` methods of a not `constexpr` constructible object?